python/tests/core/test_arrow_flight_client.py

```diff
@@ -340,3 +342,82 @@ def test_construct_query_object(self, mocker, backend_fixtures):
 
         diff = self._find_diff(query_object, query_object_reference)
         assert diff == {}
+
+    def test_construct_query_object_datetime_filter(self, mocker, backend_fixtures):
+        # Arrange
+        self._arrange_engine_mocks(mocker, backend_fixtures)
+        json1 = backend_fixtures["feature_group"]["get"]["response"]
+        test_fg1 = feature_group.FeatureGroup.from_response_json(json1)
+        mocker.patch("hsfs.constructor.query.Query.to_string", return_value="")
+        mocker.patch("hsfs.constructor.query.Query._to_string", return_value="")
+        query = test_fg1.select_all().filter(
+            test_fg1.features[0]
+            > datetime.datetime.strptime("2011-03-01 12:57:02", "%Y-%m-%d %H:%M:%S")
+        )
+
+        # Act
+        query_object = arrow_flight_client.get_instance()._construct_query_object(
+            query, "SELECT * FROM..."
+        )
+
+        # Assert
+        query_object_reference = {
+            "query_string": "SELECT * FROM...",
+            "featuregroups": {15: "test.fg_test_1"},
+            "features": {"test.fg_test_1": ["intt", "stringt"]},
+            "filters": {
+                "type": "logic",
+                "logic_type": "SINGLE",
+                "left_filter": {
+                    "type": "filter",
+                    "condition": "GREATER_THAN",
+                    "value": "2011-03-01 12:57:02",
+                    "feature": "test.fg_test_1.intt",
+                    "numeric": False,
+                },
+                "right_filter": None,
+            },
+        }
+
+        diff = self._find_diff(query_object, query_object_reference)
```

Comment:
why do we need `_find_diff` method it might not work and cause the test to fail/succeed when it shouldn't (basically forcing us to test that method as well)? cant we just use `==` or something like `DeepDiff`?

Reply:
yep think I could simplify it